File: tests/convert.php
<?php
require_once('../transcoder.php');
transcoder_set_dir('../transcoder');
$dir1 = transcoder_get_dir();
echo "transcoder directory = $dir1\n";

$filein = $argv[1];
$fileout = $argv[2];
$tranin = $argv[3];
$tranout = $argv[4];

$fp = fopen($filein,"r") or die ("cannot open $filein\n");
$fpout = fopen ($fileout,"w") or die("cannot open $fileout\n");
$n=0;
$nout=0;
$ntext=0;
 while (!feof($fp)) {
  $x = fgets($fp);
  $x = trim($x);
  if ($x == '') {continue;}
  $n++; 
  if (! preg_match('/^([^ ]+) (.+)$/',$x,$matches)) {
   echo "line $n is unknown line: '$x'\n";
   exit(1);
  }
  $head = $matches[1];
  $body = $matches[2];
  // Smith seems to use the vertical bar '|' to inhibit sandhi.
  $body = preg_replace('/\|/',' # ',$body); 
  $body = preg_replace('/ +/',' ',$body);
  $body1 = transcoder_processString($body,$tranin,$tranout);

  $y = "$head $body1";
  fwrite($fpout,"$y\n");
//  if ($n > 100) {break;}
}
fclose ($fp);
fclose($fpout);
echo "$n lines converted\n";
?>

File: tests/convert.py
"""convert.py
  Python example of transcoding
"""
import sys,codecs,re
sys.path.append('../')
import transcoder
transcoder.transcoder_set_dir('../transcoder');

def convert(filein,fileout,tranin,tranout):
 fp = codecs.open(filein,"r",'utf-8')
 fpout = codecs.open(fileout,"w",'utf-8')
 n=0;
 for x in fp:
  x = x.rstrip('\r\n')
  if (x == ''):
   continue
  n=n+1
  m = re.search(r'^([^ ]+) (.+)$',x)
  if not m:
   out = "line %s is unknown: %s" %(n,x)
   exit(1)
  head = m.group(1)
  body = m.group(2)
  #body = re.sub('/\|/',' # ',body); 
  #body = preg_replace('/ +/',' ',body);
  body1 = transcoder.transcoder_processString(body,tranin,tranout)
  y = "%s %s" % (head,body1)
  fpout.write("%s\n" % y)
 fp.close()
 fpout.close()
 print n,"lines converted\n"
#-----------------------------------------------------
if __name__=="__main__":
 filein = sys.argv[1]
 fileout = sys.argv[2]
 tranin = sys.argv[3]
 tranout = sys.argv[4]
 convert(filein,fileout,tranin,tranout)


